extmod/modbluetooth.c

```diff
@@ -172,6 +172,37 @@ STATIC void bluetooth_uuid_print(const mp_print_t *print, mp_obj_t self_in, mp_p
     }
 }
 
+mp_int_t bluetooth_uuid_get_buffer(mp_obj_t self_in, mp_buffer_info_t *bufinfo, mp_uint_t flags) {
+    mp_obj_bluetooth_uuid_t *self = MP_OBJ_TO_PTR(self_in);
+
+    if (flags == MP_BUFFER_READ) {
+        uint8_t *buf = m_new(uint8_t, self->type);
```

Comment:
get_buffer usually (I think always?) returns just a reference into the data, not a copy of it.  For efficiency reasons, and the caller always knows that it's just a reference (and for writable buffers allows the caller to modify it inplace).

So, for the 128-bit UUID it's simple to just do `bufinfo->buf = self->uuid._128`.

For 16-bit and 32-bit it might be possible to just return the address of the `_16` or `_32` entry.. but that probably won't be portable across archs with different endianness.

But looking at the code, it might be possible to simplify the UUID class a lot by always making it store the UUID value in a `uint8_t` array.  For 16 and 32 bit values it'll sometimes need to pack/unpack the value, but I think it'll simplify a lot of code paths that use UUID's, especially the one here.